docs/scripts/getTypeScriptProjects.ts

```diff
@@ -100,12 +166,71 @@ export const getTypeScriptProjects = () => {
     createProject({
       name: 'x-data-grid',
       rootPath: path.join(workspaceRoot, 'packages/grid/x-data-grid'),
-      tsConfigPath: 'tsconfig.json',
-      entryPointPath: 'src/index.ts',
-      componentsFolder: 'src/components',
-      otherComponentFiles: ['src/DataGrid/DataGrid.tsx'],
+      documentationFolderName: 'data-grid',
+      getComponentsWithPropTypes: getComponentPaths({
+        folders: ['src/components'],
+        files: ['src/DataGrid/DataGrid.tsx'],
+      }),
+      getComponentsWithApiDoc: getComponentPaths({
+        files: ['src/DataGrid/DataGrid.tsx'],
+      }),
     }),
   );
 
+  projects.set(
+    'x-data-grid-pro',
+    createProject({
+      name: 'x-data-grid-pro',
+      rootPath: path.join(workspaceRoot, 'packages/grid/x-data-grid-pro'),
+      documentationFolderName: 'data-grid',
+      getComponentsWithPropTypes: getComponentPaths({
+        folders: ['src/components'],
+        files: ['src/DataGridPro/DataGridPro.tsx'],
+      }),
+      getComponentsWithApiDoc: getComponentPaths({
+        files: ['src/DataGridPro/DataGridPro.tsx'],
+      }),
+    }),
+  );
+
+  projects.set(
+    'x-data-grid-generator',
+    createProject({
+      name: 'x-data-grid-generator',
+      rootPath: path.join(workspaceRoot, 'packages/grid/x-data-grid-generator'),
+      documentationFolderName: 'data-grid',
+    }),
+  );
```

Comment:
Why adding the data grid generator to the documented projects?

Reply:
To generate the export JSON
It is not related to the pickers at all.
But after a few months with those `package.exports.json`, I use them a lot to now what I start / stop exporting.

Reply:
I was thinking that we don't want to document this package. But yes, it does not hurt to know what is modified in the export

Reply:
Adding it to the project has no added any new content except the export file.
In the future, we could add keys like `documentSelectors: false` if the package exported selectors and we did not want to document them for instance.